ankihub/main/importing.py

```diff
@@ -187,7 +190,106 @@ def import_ankihub_deck(
+
+            # TODO Allow extra local fields
+            if not field_tuples(local_note_type["flds"]) == field_tuples(
+                remote_note_type["flds"]
+            ):
+                LOGGER.info(
+                    "Fields of local note type differ from remote note type.",
+                    local_note_type_name=local_note_type["name"],
+                    local_fields=field_tuples(local_note_type["flds"]),
+                    remote_fields=field_tuples(remote_note_type["flds"]),
+                )
+                note_types_with_field_conflicts.append(
+                    (local_note_type, remote_note_type)
+                )
+
+        if self._raise_if_full_sync_required and note_types_with_field_conflicts:
+            raise ChangesRequireFullSyncError(
+                affected_note_type_ids=set(
+                    remote_note_type["id"]
+                    for _, remote_note_type in note_types_with_field_conflicts
+                )
+            )
+
+        for local_note_type, remote_note_type in note_types_with_field_conflicts:
+            local_note_type["flds"] = _adjust_field_ords(
+                local_note_type["flds"], remote_note_type["flds"]
+            )
+            aqt.mw.col.models.update_dict(local_note_type)
+            LOGGER.info(
+                "Fields after updating the note type",
+                fields=field_tuples(
+                    aqt.mw.col.models.get(local_note_type["id"])["flds"]
+                ),
+            )
+
+    def _update_templates_and_css(
+        self, remote_note_types: Dict[NotetypeId, NotetypeDict]
+    ) -> None:
+        anking_note_types_addon_installed = is_anking_note_types_addon_installed()
+        projekt_anki_note_types_addon_installed = (
+            is_projektanki_note_types_addon_installed()
+        )
+
+        for mid, remote_note_type in remote_note_types.items():
+            local_note_type = aqt.mw.col.models.get(mid)
+
+            # We don't use new templates and css of AnKing note types if the AnKing note types addon is installed.
+            # The AnKing note types addon will handle updating the templates, while preserving the
+            # user's customizations.
+            # The same applies to ProjektAnki note types and the ProjektAnki note types addon.
+            use_new_templates_and_css = not (
+                (
+                    "anking" in remote_note_type["name"].lower()
+                    and anking_note_types_addon_installed
+                )
+                or (
+                    "projektanki" in remote_note_type["name"].lower()
+                    and projekt_anki_note_types_addon_installed
+                )
+            )
+
+            if (
+                self._raise_if_full_sync_required
+                and use_new_templates_and_css
+                and len(local_note_type["tmpls"]) != len(remote_note_type["tmpls"])
+            ):
+                raise ChangesRequireFullSyncError(
```

Comment:
Nitpick: can we move this outside the loop so all types with differing templates are reported all at once (If I'm reading this correctly)?

Reply:
Yes, I moved it out of the loop now. 

> so all types with differing templates are reported all at once

(Previously only one of the note types with differing templates would be reported, but once the user runs the full sync all of them would be updated.)